### ISSUE TITLE
Error in rep(pd$midpoints, pd$freq) : invalid 'times' argument

### ISSUE THREAD

**Issue by Ranae** [daniel1noble/metaDigitise#51]
This is most likely user error, but with newer packages, it is difficult to get advice on other platforms. 

I am trying to digitize these bar charts and have tried to go about it a number of different ways. Everything seems great until I get to this step and always get the same error message:  

    Do you want continue: 11 plots out of 12 plots remaining (y/n)  
    n
    Error in rep(pd$midpoints, pd$freq) : invalid 'times' argument

Here I am just trying to get the values as if working with a scatterplot. What am I doing wrong?
![image](https://user-images.githubusercontent.com/10119458/37720025-621dde82-2cf4-11e8-9221-2cd57990ae99.png)

**Comment by joelpick** [comment]
Hi Ranae. Thanks sharing this - we are very keen to identify bugs. Are any of the other 10 plots digitised histograms? This seems like an error to do with that, rather than the scatterplot (midpoints and freq are generated internally only for histograms) histograms? Without seeing the other plots it might be hard to diagnose - would you be able to upload the folder of images etc?

**Comment by Ranae** [comment]
There's a variety in the folder - multiple panels from two figures.

Let me know if you can't get the folder [here](https://drive.google.com/open?id=1YFCLYdJqH5yD2rfD-OkFo5aIlQ88ozvy).

**Comment by joelpick** [comment]
Thanks for that - I think the problem is with `Fig1_kgN2O-N_ha_yr` - when running `metaDigitise()` if you go to "3: Edit existing data", then "2: Choose specific file to edit" then choose "Fig1_kgN2O-N_ha_yr" you can see there is a problem (I found this by instead choosing "1: Cycle through images" and eyeballing all of them). As you have done in the example above, I think that treating these plots as scatterplots maybe the best way, although I guess it depends what you want from them.

**Comment by joelpick** [comment]
Out of interest, what are you running this on? (operating system and R) Do the points appear as you are clicking, or once you have finished? from redrawing Fig1_kgN2O-N_ha_yr it seems as thought these may not have been appearing as you clicked on the points?

**Comment by Ranae** [comment]
I am using Windows 10 and R version 3.4.3 (2017-11-30) -- "Kite-Eating Tree". The points did appear when I was clicking. I think on the first one (Fig1_kgN2O-N_ha_yr) I was a little too excited, didn't read the instructions, and did some weird stuff. Then I realized it would be better to split panels and also, "draw lines" across the tops of the bar. 

Let me give this another go without the multi-panels files in the way.

**Comment by Ranae** [comment]
Looks like I'm rolling here! I went ahead and deleted my whole caldat folder and started from scratch with no multipanel pngs in the folder. I think you are right that my first sloppy try did something to mess up the rest of the process. 

Now that you mention it, I didn't see my last point while clicking, but it showed up after clicking the red box.  But I think you can probably switch this issue from "bug" to "cautionary tale".

**Comment by daniel1noble** [comment]
Sorry Ranae and Joel for the slow reply. Glad Joel sorted this. Thanks Joel!

–––
Dr. Daniel Noble | ARC DECRA Fellow
Level 5 West, Biological Sciences Building (E26)
Ecology & Evolution Research Centre (E&ERC)
School of Biological, Earth and Environmental Sciences (BEES)
*The University of New South Wales*
Sydney, NSW 2052
AUSTRALIA

T : +61 430 290 053
E : daniel.noble@unsw.edu.au <daniel.noble@mq.edu.au>
W: www.nobledan.com
Github: https://github.com/daniel1noble

**Comment by joelpick** [comment]
Glad thats fixed. I think it points out a good place for error catching and warning messages though. We should be able to make it warn you which image is causing the problem and return the 'good' data. Thanks for sharing!!

**Comment by daniel1noble** [comment]
Agreed! This is a good idea and should be easy to implement. We will make this happen.
